lib/utils.py

```diff
@@ -57,14 +57,24 @@ def validate_url(string):
 
 def get_node_ip():
     if arch in ('armv6l', 'armv7l'):
-        active_interfaces = interfaces()
 
-        if 'eth0' in active_interfaces:
-            interface = 'eth0'
-        elif 'wlan0' in active_interfaces:
-            interface = 'wlan0'
-        else:
-            raise Exception("No active network interface found.")
+        interface = None
+        try:
+            file_carrier = open('/sys/class/net/eth0/carrier')
+            file_operstate = open('/sys/class/net/eth0/operstate')
+
+            if "1" in file_carrier.read() and "up" in file_operstate.read():
+                interface = 'eth0'
```

Comment:
Actually, take a look at this:
https://github.com/Screenly/screenly-ose/blob/master/ansible/roles/network/files/screenly_net_mgr.py#L157-L162